data/pools/transactionPool.go

```diff
@@ -360,34 +367,24 @@ func (pool *TransactionPool) Lookup(txid transactions.Txid) (tx transactions.Sig
 // returned success).  This is used as an optimization to avoid
 // re-checking signatures on transactions that we have already
 // verified.
-func (pool *TransactionPool) Verified(txn transactions.SignedTxn) bool {
+func (pool *TransactionPool) Verified(txn transactions.SignedTxn, params verify.Params) bool {
 	if pool == nil {
 		return false
 	}
 	pool.pendingMu.RLock()
 	defer pool.pendingMu.RUnlock()
-	pendingSigTxn, ok := pool.pendingTxids[txn.ID()]
+	cacheval, ok := pool.pendingTxids[txn.ID()]
 	if !ok {
 		return false
 	}
 
+	if cacheval.params != params {
+		return false
+	}
+	pendingSigTxn := cacheval.txn
 	return pendingSigTxn.Sig == txn.Sig && pendingSigTxn.Msig.Equal(txn.Msig) && pendingSigTxn.Lsig.Equal(&txn.Lsig)
```

Comment:
Probably ok but others should probably think about it too:

We only compare the `verify.Params`, not the entire `Context`. This means that you could pass `Verified` a `verify.Params` from inside a `Context` for which the group index / group would cause a logic sig to now fail.

I think this is probably OK because we only add transactions via `transactionGroup`, which does a well formedness check on the group (so you shouldn't be able to change the other parts of `Context` without also changing the `txid`)